blivet/devices/lvm.py

```diff
@@ -1901,7 +1901,7 @@ def _set_size(self, newsize):
     @type_specific
     def max_size(self):
         """ The maximum size this lv can be. """
-        max_lv = self.size + self.vg.free_space
+        max_lv = self.vg.align(self.size, roundup=True) + self.vg.align(self.vg.free_space)
```

Comment:
This definitely needs some explanation -- how changing filesystem minimal size influences max size of an LV?

Reply:
Probably some rogue line of code. Removed.

Reply:
The changes to the tests seem to have revealed a bug in the lv max_size calculation. If the lv size is unaligned, you end up with a bit of extra space available that isn't accounted for in max_size. In the test I was doing, the pv was 1 GiB and the lv was 499MiB. The vg should have had 520GiB free (1GiB - 1 extent - lv size), which it did. It was reporting a max_size of 1019MiB (lv size + vg free). However, there is the remaining 1MiB in the lv's last extent that should be taken into account as well, making the max_size for the lv 1020MiB. The current/upstream code was reporting it as 1019 instead. This should be split out to a separate commit with a good explanation in the commit message.

Reply:
I have split PR to two commits. The second one describes the uncovered issue and fixes it.